src/lib/protocols/bittorrent.c

```diff
@@ -531,7 +545,11 @@ static void ndpi_search_bittorrent(struct ndpi_detection_module_struct *ndpi_str
 
 	  if(is_utpv1_pkt(packet->payload, packet->payload_packet_len)) {
 	    bt_proto = ndpi_strnstr((const char *)&packet->payload[20], BITTORRENT_PROTO_STRING, packet->payload_packet_len-20);
-	    goto bittorrent_found;
+	    if(flow->packet_counter > 2) {
```

Comment:
Would make sense something like `if(flow->packet_counter > 2 || bt_proto) {` ?
If we find the BitTorrent string, maybe we can stop here. What do you think?